model/riscv_pmp_control.sail

```diff
@@ -152,104 +158,39 @@ function pmpMatchEntry(addr: xlenbits, width: xlenbits, acc: AccessType(ext_acce
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp12cfg, pmpaddr12, pmpaddr11) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp13cfg, pmpaddr13, pmpaddr12) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp14cfg, pmpaddr14, pmpaddr13) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp15cfg, pmpaddr15, pmpaddr14) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue => match priv {
-                      Machine => true,
-                      _       => false
-                    }
-  }}}}}}}}}}}}}}}};
-
-  if   check
-  then None()
-  else match acc {
-    Read(_)      => Some(E_Load_Access_Fault()),
-    Write(_)     => Some(E_SAMO_Access_Fault()),
-    ReadWrite(_) => Some(E_SAMO_Access_Fault()),
-    Execute()    => Some(E_Fetch_Access_Fault())
-  }
+
+  foreach (i from 0 to 63) {
+    let prev_pmpaddr = (if i > 0 then pmpReadAddrReg(i) else zeros());
+    match pmpMatchEntry(addr, width, acc, priv, pmpcfg_n[i], pmpReadAddrReg(i), prev_pmpaddr) {
+      PMP_Success  => { return None(); },
+      PMP_Fail     => { return Some(accessToFault(acc)); },
+      PMP_Continue => (),
+    }
+  };
+  if priv == Machine then None() else Some(accessToFault(acc))
 }
 
 function init_pmp() -> unit = {
-  pmp0cfg  = update_A(pmp0cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp1cfg  = update_A(pmp1cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp2cfg  = update_A(pmp2cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp3cfg  = update_A(pmp3cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp4cfg  = update_A(pmp4cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp5cfg  = update_A(pmp5cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp6cfg  = update_A(pmp6cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp7cfg  = update_A(pmp7cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp8cfg  = update_A(pmp8cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp9cfg  = update_A(pmp9cfg,  pmpAddrMatchType_to_bits(OFF));
-  pmp10cfg = update_A(pmp10cfg, pmpAddrMatchType_to_bits(OFF));
-  pmp11cfg = update_A(pmp11cfg, pmpAddrMatchType_to_bits(OFF));
-  pmp12cfg = update_A(pmp12cfg, pmpAddrMatchType_to_bits(OFF));
-  pmp13cfg = update_A(pmp13cfg, pmpAddrMatchType_to_bits(OFF));
-  pmp14cfg = update_A(pmp14cfg, pmpAddrMatchType_to_bits(OFF));
-  pmp15cfg = update_A(pmp15cfg, pmpAddrMatchType_to_bits(OFF))
+  assert(
+    sys_pmp_count() == 0 | sys_pmp_count() == 16 | sys_pmp_count() == 64,
+    "sys_pmp_count() must be 0, 16, or 64"
+  );
+
+  foreach (i from 0 to 63) {
+    // On reset the PMP register's A and L bits are set to 0 unless the plaform
+    // mandates a different value.
+    pmpcfg_n[i] = update_A(pmpcfg_n[i], pmpAddrMatchType_to_bits(OFF));
```

Comment:
Ah nice, I didn't realise `update_` was the "old" version. Updated.

Reply:
I also updated the similar pattern in `pmpWriteCfg()` - you might want to check there (and nitpick my `if`/`then` formatting!) :-D

Reply:
@arichardson ,  since you originated this comment,  would you mark it as resolved?  (That is,  if you think it's resolved.) 

As a general rule (based on the feedback at the golden-model meeting this morning),  we'd like for the originator of the comment to mark it as resolved.   Thanks.

Reply:
It appears you need repository write access or be the creator of the PR to resolve comments, so I can't do it, but as far as I can tell this has been resolved.

Reply:
Per @arichardson comment,  I'm resolving this conversation.